arnold/plugins/gaffer.mtd

```diff
@@ -340,1137 +203,602 @@
 
 
-[node aovWritePoint]
+[node ramp_float]
 
-	gaffer.nodeMenu.category STRING "Maya/General Utilities"
+	gaffer.nodeMenu.category STRING "Utility"
 
 
-[node aovWritePoint2]
+[node ramp_rgb]
 
-	gaffer.nodeMenu.category STRING "Maya/General Utilities"
+	gaffer.nodeMenu.category STRING "Utility"
 
 
-[node aovWriteVector]
+[node random]
 
-	gaffer.nodeMenu.category STRING "Maya/General Utilities"
+	primaryInput STRING "input_color"
+	gaffer.nodeMenu.category STRING "Maths"
 
 
-[node cameraUvMapper]
+[node range]
 
-	gaffer.nodeMenu.category STRING "Maya/General Utilities"
+	primaryInput STRING "input"
+	gaffer.nodeMenu.category STRING "Maths"
 
 
-[node colorToFloat]
+[node ray_switch_rgba]
 
-	gaffer.nodeMenu.category STRING "Maya/Color Utilities"
+	gaffer.nodeMenu.category STRING "Utility"
 
 
-[node mayaBump2D]
+[node ray_switch_shader]
 
-	gaffer.nodeMenu.category STRING "Maya/General Utilities"
+	gaffer.nodeMenu.category STRING "Utility"
 
 
-[node mayaFluid]
+[node reciprocal]
 
-	gaffer.nodeMenu.category STRING "Maya/3D Textures"
+	primaryInput STRING "input"
+	gaffer.nodeMenu.category STRING "Maths"
 
 
-[node mayaMarble]
+[node rgb_to_float]
 
-	gaffer.nodeMenu.category STRING "Maya/3D Textures"
+	gaffer.nodeMenu.category STRING "Utility"
 
 
-[node meshLightMaterial]
+[node rgb_to_rgba]
 
-	gaffer.nodeMenu.category STRING "Maya/Surface"
+	gaffer.nodeMenu.category STRING "Utility"
 
 
-[node point2ToFloat]
+[node rgb_to_vector]
 
-	gaffer.nodeMenu.category STRING "Maya/General Utilities"
+	gaffer.nodeMenu.category STRING "Utility"
 
 
-[node pointToFloat]
+[node rgba_to_float]
 
-	gaffer.nodeMenu.category STRING "Maya/General Utilities"
+	gaffer.nodeMenu.category STRING "Utility"
 
 
-[node shadowCatcher]
+[node shadow_matte]
 
-	gaffer.nodeMenu.category STRING "Maya/Surface"
+	gaffer.nodeMenu.category STRING "Utility"
 
 
-[node skin]
+[node shuffle]
 
-	gaffer.nodeMenu.category STRING "Maya/Surface"
+	gaffer.nodeMenu.category STRING "Utility"
 
 
-[node userDataBool]
+[node sign]
 
-	gaffer.nodeMenu.category STRING "Maya/General Utilities"
+	primaryInput STRING "input"
+	gaffer.nodeMenu.category STRING "Maths"
 
 
-[node userDataColor]
+[node skin]
 
-	gaffer.nodeMenu.category STRING "Maya/General Utilities"
+	# DEPRECATED
+	gaffer.nodeMenu.category STRING ""
 
 
-[node userDataFloat]
+[node sky]
 
-	gaffer.nodeMenu.category STRING "Maya/General Utilities"
+	# DEPRECATED
+	gaffer.nodeMenu.category STRING ""
 
 
-[node userDataInt]
+[node space_transform]
 
-	gaffer.nodeMenu.category STRING "Maya/General Utilities"
+	gaffer.nodeMenu.category STRING "Maths"
 
 
-[node userDataPnt2]
+[node sqrt]
 
-	gaffer.nodeMenu.category STRING "Maya/General Utilities"
+	primaryInput STRING "input"
+	gaffer.nodeMenu.category STRING "Maths"
 
 
-[node userDataString]
+[node standard]
 
-	gaffer.nodeMenu.category STRING "Maya/General Utilities"
+	# DEPRECATED
+	gaffer.nodeMenu.category STRING ""
 
 
-[node userDataVector]
+[node standard_surface]
```

Comment:
What does the NodeEditor UI for standard_surface look like now? Is it reasonably arranged into sections like the AlSurface was?

Reply:
Nope, it's a huge mess.  To get it looking nice, I took an mtd for it that ships with htoa, and modified a bunch of things to make it work better with Gaffer, and rewrote some tooltips to make more sense - currently I've only released this internally with IERendering.  Do you think I should be trying to ship this with Gaffer?  I guess I'd have to check with the SolidAngle folks that they're OK with me modifying and redistributing a file that comes with htoa.  ( Also, the syntax of mtd files for Houdini has changed, so if we want to properly be compatible with current mtd files, I guess we should update our mtd parser.  Now that there are only a handful of Arnold shaders we actually want to use, manually rewriting mtd files is quicker than updating the parser, but maybe you feel like this is something we should support in public Gaffer )

Reply:
Whatever we do, Gaffer needs to be useable out of the box with nothing more than Arnold core. I think there's a good case to be made that Solid Angle should ship a useable .mtd file with Arnold core, since that's where the shaders are now. I guess they'd be reluctant to include things like the crazy Houdini grouping syntax though unless this has improved. Maybe you could broach the subject with them? Perhaps a compromise is that they ship sensible documentation strings and presets with Arnold core, leaving the crazy weird stuff in HtoA, and then we fill in with some section metadata of our own that maps more directly to Gaffer layouts?